doc/development/plugins.rst

```diff
@@ -1,397 +1,398 @@
-      and the corresponding operation in the targeted framework, ensure that the
-      conversion between the two conventions takes place automatically
-      by the plugin device.
+For example:
 
-* :func:`pennylane.Device.capabilities`: A class method which returns the dictionary of capabilities of a device. A
-  new device should override this method to retrieve the parent classes' capabilities dictionary, make a copy
-  and update and/or add capabilities before returning the copy.
+.. code-block:: python
 
-  Examples of capabilities are:
+    class MyDevice(Device):
+        """My Documentation."""
 
-  * ``'model'`` (*str*): either ``'qubit'`` or ``'cv'``.
+        def execute(self, circuits: QuantumScriptOrBatch, execution_config: "ExecutionConfig" = DefaultExecutionConfig):
+            return 0.0 if isinstance(circuits, qml.tape.QuantumScript) else tuple(0.0 for c in circuits)
 
-  * ``'returns_state'`` (*bool*): ``True`` if the device returns the quantum state via ``dev.state``.
+    dev = MyDevice()
 
-  * ``'supports_inverse_operations'`` (*bool*): ``True`` if the device supports
-    applying the inverse of operations. Operations which should be inverted
-    have the property ``operation.inverse == True``.
+    @qml.qnode(dev)
+    def circuit():
+        return qml.state()
 
-  *  ``'supports_tensor_observables'`` (*bool*): ``True`` if the device supports observables composed from tensor
-     products such as ``PauliZ(wires=0) @ PauliZ(wires=1)``.
+    circuit()
 
-  *  ``'supports_tracker'`` (*bool*): ``True`` if it has a device tracker attribute and updates information with
-     it.
+This execute method works in tandem with the optional ``Device.preprocss``, described below in more detail.
+Preprocessing turns generic circuits into ones supported by the device, or raises an error if the circuit is invalid. Execution then
+turns those supported circuits into numerical results. 
 
-  Some capabilities are queried by PennyLane core to make decisions on how to best run computations, others are used
-  by external apps built on top of the device ecosystem.
+In a more minimal example, for any initial batch of quantum tapes and a config object, we expect to be able to do:
 
-  To find out which capabilities are (possibly automatically) defined for your device ``dev = device('my.device')``,
-  check the output of ``dev.capabilities()``.
+.. code-block:: python
 
-Adding arguments to your device
---------------------------------
+    transform_program, execution_config = dev.preprocess(initial_config)
+    circuit_batch, postprocessing = transform_program(initial_circuit_batch)
+    results = dev.execute(circuit_batch, execution_config)
+    final_results = postprocessing(results)
 
-.. important::
 
-    PennyLane supports both qubit and continuous-variable (CV) devices. However, from
-    here onwards, we will demonstrate plugin development focusing on qubit-based devices
-    inheriting from the :class:`~.QubitDevice` class.
+Shots
```

Comment:
It just came to me. We might add a comment here about how `shots.total_shots` being `None` corresponds to analytic execution, and that the device might want to raise an error if it is not supported.